src/racer/core.rs

```diff
@@ -738,6 +739,8 @@ pub struct Session<'c> {
     /// The file cache is used within a session to prevent multiple reads. It is
     /// borrowed here in order to support reuse across Racer operations.
     cache: &'c FileCache,
+    /// Cache for crate roots
+    cargo_roots: RefCell<HashMap<(String, path::PathBuf), Option<path::PathBuf>>>,
```

Comment:
There is interest in having the `Session` type be sync. Are you open to changing this to a `RwLock` instead?

Reply:
That's your call of course. (Maybe adding a test would be good to keep it `Sync`?)

Hmm, if Racer stuff is run in different threads, it might also be better to make the TOML cache an `Arc<Mutex>` lazy-static instead of a thread local?

Reply:
`Session` is currently not `Sync` (and hasn't been for a long time, because `FileCache` isn't). If you want to make that change, I'd propose to do it separately.